File: solutions/lua/darts/1/darts.lua
local Darts = {}

function Darts.score(x, y)
    local dist = math.sqrt(x ^ 2 + y ^ 2)

    if dist <= 1 then
        return 10
    elseif dist <= 5 then
        return 5
    elseif dist <= 10 then
        return 1
    end

    return 0
end

return Darts

File: solutions/go/darts/1/darts.go
// Package darts helps calculate scores for the game of darts.
package darts

import "math"

// Score returns the score earned by a dart throw.
func Score(x, y float64) int {
	switch dist := math.Sqrt(math.Pow(x, 2) + math.Pow(y, 2)); {
	case dist <= 1:
		return 10
	case dist <= 5:
		return 5
	case dist <= 10:
		return 1
	default:
		return 0
	}
}
